src/BatchDealImportService.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\amazon_product_widget;
+
+use Drupal\amazon_product_widget\Exception\DealFeedFinishedProcessingException;
+
+/**
+ * Class BatchDealImportService.
+ *
+ * @package Drupal\amazon_product_widget
+ */
+class BatchDealImportService {
+
+  /**
+   * Imports deals into storage from CSV.
+   *
+   * @param string $filename
+   *   Filename from where to import.
+   * @param int $total
+   *   The total entries in the CSV file.
+   * @param $context
+   *   The context.
+   */
+  public static function importChunked(string $filename, int $total, &$context) {
+    /** @var \Drupal\amazon_product_widget\DealFeedService $dealFeedService */
+    $dealFeedService = \Drupal::service('amazon_product_widget.deal_feed_service');
+    $maxProcessingTime = $dealFeedService->getMaxProcessingTime();
+
+    // How many imports to do per call to DealFeedService::import().
+    $importsPerRound = 1000;
+
+    if (empty($context['sandbox'])) {
```

Comment:
sandbox can be empty? i always checked for a key i set mayself in init, eg. $context['sandbox']['total']